app/scripts/directives/stlCourtsDatepicker.dir.js

```diff
@@ -0,0 +1,372 @@
+'use strict';
+/*
+    Usage: <stl-courts-datepicker date-format="some format string"></stl-courts-datepicker>
+    <stl-courts-datepicker date-field="ctrl.dob" date-valid="ctrl.dobValid" date-over-18="ctrl.dobOver18" date-format="MM/DD/YYYY"></stl-courts-datepicker>
+*/
+
+angular.module('yourStlCourts').directive('stlCourtsDatepicker', function($timeout,moment){
+  return{
+    restrict: 'E',
+    scope:{
+      dateField:'=',
+      dateValid:'=',
+      dateOver18:'='
+    },
+    templateUrl: 'views/stlCourtsDatepicker.html',
+    controller: 'StlCourtsDatepicker as ctrl',
+    link: function($scope,element,attrs){
```

Comment:
You have a bunch of custom interaction code in this directive. And while it's good that it's in a directive, you're bypassing several things that angular gives you in the HTML to trim this up. Just because you need to do things on the event object doesn't mean you can't do it through angular. I'm not requesting you change anything - just pointing out that less custom event/interaction code, the better in angular